hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1401,6 +1403,25 @@ public void destroy() {
         clearPartition(false, true);
     }
 
+    @Override
+    public void replaceAll(BiFunction function) {
+        Iterator<Map.Entry<Data, Record>> entryIterator = storage.mutationTolerantIterator();
```

Comment:
Anything that `replace` method does I think should be done here as well.

A call to `checkIfLoaded` 
A call to `evictIfExpired` 
A call to `accessRecord` 
And update the record with `DefaultRecordStore.updateRecord` to make finishing calls. 

Also thinking about it, we should not call user functions on partition threads. A core team member help us here may be. I will ask for a review for this. 
In that case, we might be forced to use a similar thing used in MapProxy to simplify the solution. 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java#L1422

But we can not do the same on the ClientMapProxy because we don't have the cluster version in the client. 
We might try to do the same from ReplaceAlMessageTask which is 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java#L1293
To make it clear, to reach this line, I started from here 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java#L1422

Reply:
As discussed, implemented the feature using processor instead of creating new operation.